chia/data_layer/data_layer.py

```diff
@@ -86,38 +93,72 @@ def server(self) -> ChiaServer:
-        self,
+    @property
+    def data_store(self) -> DataStore:
+        # This is a stop gap until the class usage is refactored such the values of
+        # integral attributes are known at creation of the instance.
+        if self._data_store is None:
+            raise RuntimeError("data_store not assigned")
+
+        return self._data_store
+
+    @property
+    def wallet_rpc(self) -> WalletRpcClient:
+        # This is a stop gap until the class usage is refactored such the values of
+        # integral attributes are known at creation of the instance.
+        if self._wallet_rpc is None:
+            raise RuntimeError("wallet_rpc not assigned")
+
+        return self._wallet_rpc
+
+    @property
+    def subscription_lock(self) -> asyncio.Lock:
+        # This is a stop gap until the class usage is refactored such the values of
+        # integral attributes are known at creation of the instance.
+        if self._subscription_lock is None:
+            raise RuntimeError("subscription_lock not assigned")
+
+        return self._subscription_lock
+
+    @classmethod
+    def create(
+        cls,
         config: Dict[str, Any],
         root_path: Path,
         wallet_rpc_init: Awaitable[WalletRpcClient],
         downloaders: List[str],
         uploaders: List[str],  # dont add FilesystemUploader to this, it is the default uploader
         name: Optional[str] = None,
-    ):
+    ) -> DataLayer:
         if name == "":
             # TODO: If no code depends on "" counting as 'unspecified' then we do not
             #       need this.
             name = None
-        self.initialized = False
-        self.config = config
-        self.root_path = root_path
-        self.connection = None
-        self.wallet_rpc_init = wallet_rpc_init
-        self.log = logging.getLogger(name if name is None else __name__)
-        self._shut_down: bool = False
-        db_path_replaced: str = config["database_path"].replace("CHALLENGE", config["selected_network"])
-        self.db_path = path_from_root(self.root_path, db_path_replaced)
-        self.db_path.parent.mkdir(parents=True, exist_ok=True)
+
         server_files_replaced: str = config.get(
             "server_files_location", "data_layer/db/server_files_location_CHALLENGE"
         ).replace("CHALLENGE", config["selected_network"])
-        self.server_files_location = path_from_root(self.root_path, server_files_replaced)
+
+        db_path_replaced: str = config["database_path"].replace("CHALLENGE", config["selected_network"])
+
+        self = cls(
+            config=config,
+            root_path=root_path,
+            wallet_rpc_init=wallet_rpc_init,
+            log=logging.getLogger(name if name is None else __name__),
+            _shut_down=False,
+            db_path=path_from_root(root_path, db_path_replaced),
+            server_files_location=path_from_root(root_path, server_files_replaced),
+            none_bytes=bytes32([0] * 32),
+            _server=None,
```

Comment:
```suggestion
```

There is no default yet but seems like you could just assign it and drop this.

Reply:
Could we leave this?  It looks to be pointless but also looks to be part of a common pattern that we should maybe analyse.  For example, the wallet probably shouldn't need it either.  Also the timelord etc as related to my PRs around removing the unused peer ports in the config.

Reply:
Not sure what the common pattern is you are referring to here or what your plans are with keeping this, but im ok if you see any sense in it. So feel free to close it but i want to mention, there are other classes where we already assign a default value via the dataclass to the optional server like proposed.

Reply:
https://github.com/chia-network/chia-blockchain/commit/0cc9aa655d2eaf77e033d817aa2b20107d00dd18

I...  think I wasn't thinking straight...  sorry.